docs/userguide/core-worker.md

```diff
@@ -1,3 +1,61 @@
 # Workflow Core: Worker
 
-_Coming soon!_
+## The Role of a Worker
+
+`Worker` is a protocol (in Swift) and interface (in Kotlin) that defines an asynchronous task that
+can be performed by a `Workflow`. `Worker`s only emit outputs, they do not have a `Rendering` type.
+They are similar to child workflows with `Void`/`Unit` rendering types.
+
+A workflow can ask the infrastructure to await the result of a worker by handing that worker to the
+`RenderContext` within a call to the `render` method. A workflow can handle outputs from a `Worker`.
+
+## Workers provide a declarative window into the imperative world
```

Comment:
I’m not sure exactly where, but it may be nice to fit something about “workers are a way to perform side effects” into this paragraph somewhere. Mostly trying to give people who have some preexisting mental models something to attach them too.

Reply:
> ## Workers can perform side effects
> 
> Unlike workflows' `render` method, which can be called many times and must be idempotent, workers
> are started and then ran until completion (or cancellation) – independently of how many times the
> workflow running them is actually rendered. This means that side effects that should be performed
> only once when a workflow enters a particular state, for example, should be placed into a `Worker`
> that the workflow runs while in that state.